presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -239,6 +241,26 @@ public void recordFinish()
         finishUserNanos.getAndAdd(nanosBetween(intervalUserStart.get(), currentThreadUserTime()));
     }
 
+    /**
+     * Get a future that is completed when the {@link #notifyAsync()} method is called, or
+     * next time this method is called.  This api is designed to only be used by {@link Driver}.
```

Comment:
- why do you recreate feature on each call? To prevent mem leak in listeners being added in `firstFinishedFuture`? Some hint at this in a comment would be helpful
- capitalize `API`
- nit: do we need a comment that this is dedicated for Driver? E.g `OperatorContext#setMemoryRevocationRequestListener` is dedicated but doesn't have this comment